common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResult.java

```diff
@@ -22,7 +22,117 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
 
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResponse;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_CODE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Browser.RESPONSE_FINAL_URL;
 
 public class AzureActiveDirectoryAuthorizationResult extends AuthorizationResult {
```

Comment:
Consider creating and `extending` an intermediate subclass of `AuthorizationResult` called `MicrosoftAuthorizationResult`-- chances are, there's some opportunity for code reuse [or there will be in the future] between `AzureActiveDirectoryAuthorizationResult` and `MicrosoftStsAuthorizationResult`; that re-use should live in a Microsoft-specific superclass that exists separately of `AuthorizationResult`.